lib/mycroft_core/file.c

```diff
@@ -373,50 +380,61 @@ int mc_file_read_raw(mc_file_t* file, fsize_t amount, uint8_t* outbuf)
  * @param offset Offset into the file to start reading at.
  * @param amount Amount of bytes to read.
  * @param outbuf Buffer to place bytes in.
+ * @param amount_read Amount of bytes read.  Can be NULL.
  *
- * @return Returns 0 on success, negative value on error.
+ * @return Returns mc_error_t status.
  */
-int mc_file_read_value(mc_file_t* file, fsize_t offset, fsize_t amount, uint8_t* outbuf)
+mc_error_t mc_file_read_value(mc_file_t* file, fsize_t offset,
+                              fsize_t amount, uint8_t* outbuf,
+                              fsize_t* amount_read)
 {
+    mc_error_t rc;
+
     /* Make sure that we don't read off the end of the file */
     if ((offset + amount) >= file->size) {
         fprintf(stderr, "Tried to read off the end of the file.\n");
-        return -1;
+        return MC_FILE_ERR_BOUNDS;
     }
 
     //TODO: check to see if we can read from editor_buffer's buf?
 
     /* Save old position and move to offset */
     fsize_t old_cursor = file->cursor;
-    if (mc_file_set_cursor(file, offset) < 0) {
+    rc = mc_file_set_cursor(file, offset);
+    if (rc != MC_OK) {
         perror("DEBUG: Error in mc_file_set_cursor: ");
-        return -1;
+        return rc;
     }
 
     /* Read desired content */
-    long int res = fread(outbuf, 1, amount, file->fp);
+    size_t res = fread(outbuf, 1, amount, file->fp);
+    if (amount_read != NULL)
+        (*amount_read) = (fsize_t) res;
     if (res != amount) {
         if (ferror(file->fp)) {
             perror("DEBUG: Error in mc_file_write_pos");
-            return -1;
+            return MC_FILE_ERR_IO;
         }
         else if (feof(file->fp)) {
             fprintf(stderr, "Tried to read past end of file.\n");
-            return -1;
+            return MC_FILE_ERR_IO;
         }
         else {
-            fprintf(stderr, "fread failed for some unknown reason.  Report this!\n");
-            return -1;
+            fprintf(stderr,
```

Comment:
Shouldn't lines be kept under 80 characters long?  If I don't break that line, I break the 80 col rule.